### PR TITLE
Prepare 0.103.0 release

### PR DESCRIPTION
**Pull request by swiatekm-sumo**
I've also added the prometheus operator version to the changelog, as originally discussed in #2757.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,32 @@
...
+- `auto-instrumentation`: Fix webserver instrumentation log file name (#2978)
+  
+  Since webserver instrumentation 1.0.4, the configuration log file has been renamed from appdynamics_sdk_log4cxx.xml.template to opentelemetry_sdk_log4cxx.xml.template. The operator upgraded the webserver instrumentation version but haven't change the configuration file name.
+- `target-allocator`: Fixes a bug that didn't automatically create a PDB for a TA with per-node strategy (#2900)
+
+### Components
+
+* [OpenTelemetry Collector - v0.103.0](https://github.com/open-telemetry/opentelemetry-collector/releases/tag/v0.103.0)
+* [OpenTelemetry Contrib - v0.103.0](https://github.com/open-telemetry/opentelemetry-collector-contrib/releases/tag/v0.103.0)
+* [Prometheus Operator - v0.74.0](https://github.com/prometheus-operator/prometheus-operator/releases/tag/v0.74.0)
```
Comment (jaronoff97) [inline CHANGELOG.md:22]: i was thinking we would add this to our compat matrix in the readme not the components section given this isn't something we release but depend on

Comment (swiatekm-sumo) [reply CHANGELOG.md:22]: I thought I'd add it to the release notes so it's easily apparent, but I guess the compat matrix works as well.

File: versions.txt [hunk 2]
```diff
@@ -2,16 +2,16 @@
 # by default with the OpenTelemetry Operator. This would usually be the latest
 # stable OpenTelemetry version. When you update this file, make sure to update the
 # the docs as well.
-opentelemetry-collector=0.102.1
+opentelemetry-collector=0.103.0
```
Comment (jaronoff97) [inline versions.txt:5]: i think we want to use collector [v0.103.1](https://github.com/open-telemetry/opentelemetry-collector-releases/releases/tag/v0.103.1)

Comment (swiatekm-sumo) [reply versions.txt]: Good catch.

**Review by jaronoff97 — APPROVED** [review]

**Review by TylerHelmuth — APPROVED** [review]

**Review by pavolloffay — APPROVED** [review]

**Pull request merged**